src/dll/util.c

```diff
@@ -1,23 +1,37 @@
 #include <processthreadsapi.h>
 
-typedef BOOL (WINAPI *PSET_PROCESS_INFORMATION)(HANDLE, PROCESS_INFORMATION_CLASS, LPVOID, DWORD);
+typedef BOOL(WINAPI* PSET_PROCESS_INFORMATION)(HANDLE, PROCESS_INFORMATION_CLASS, LPVOID, DWORD);
 
-BOOL UtilSetPowerThrottlingState(ULONG feature, BOOL enable) {
+BOOL UtilSetPowerThrottlingState(ULONG feature, BOOL enable)
+{
 	static PSET_PROCESS_INFORMATION fnSetProcessInformation = NULL;
 
-	if(!fnSetProcessInformation) {
-		fnSetProcessInformation =
-			(PSET_PROCESS_INFORMATION)GetProcAddress(GetModuleHandle(L"kernel32"), "SetProcessInformation");
+	// If the SetProcessInformation function is not already loaded, attempt to load it.
+	if (!fnSetProcessInformation)
+	{
+		HMODULE hKernel32 = GetModuleHandle(L"kernel32"); // Get a handle to the kernel32 module.
```

Comment:
`kernel32` is always available on Windows (it's loaded into every process). I don't think this function can fail.